### PR TITLE
[CSL-738] Use MonadThrow, MonadCatch, MonadMask

### PR DESCRIPTION
**Pull request by int-index**

### CODE REVIEW

File: lrc/Pos/Lrc/Mode.hs [hunk 1]
```diff
@@ -22,7 +22,7 @@ type LrcMode ctx m
        , MonadGState m
        , MonadDB m
        , MonadIO m
-       , Mockables m [Async, Bracket, Concurrently, Delay]
+       , Mockables m [Async, Concurrently, Delay]
                      -- ^ alphabet for the youngest haskellers
```
Comment (gromakovsky) [inline lrc/Pos/Lrc/Mode.hs]: You've broken the alphabet :(
Can you delete this comment? ;d

File: auxx/src/Repl.hs [hunk 2]
```diff
@@ -89,10 +88,10 @@ createAuxxRepl = do
     lastResultVar <- newEmptyMVar
     (printActionVar :: MVar (PrintAction IO)) <- newEmptyMVar
     let
-        withCommand :: forall m. (MonadIO m, Mockable Catch m) => (Text -> m ()) -> m ()
+        withCommand :: forall m. (MonadIO m, MonadCatch m) => (Text -> m ()) -> m ()
         withCommand cont = do
             cmd <- takeMVar nextCommandVar
-            res <- handle
+            res <- handleAsync
```
Comment (gromakovsky) [inline auxx/src/Repl.hs]: Is it intentional or you just want to preserve existing logic and change it to `handleAny` later?
_perhaps I can figure that out by reading the code, but I don't want to ;d_

File: networking/src/Mockable/Concurrent.hs [hunk 3]
```diff
@@ -46,10 +46,11 @@ module Mockable.Concurrent (
 
   ) where
 
-import           Control.Exception (AsyncException (ThreadKilled), Exception)
+import           Control.Exception (AsyncException (..))
+import           Control.Exception.Safe hiding (throwTo)
+--import           Control.Exception (AsyncException (ThreadKilled), Exception)
```
Comment (gromakovsky) [inline networking/src/Mockable/Concurrent.hs]: First, I see implicit imports, second, is this commented out line needed?

File: networking/src/Mockable/Concurrent.hs [hunk 4]
```diff
@@ -121,13 +122,14 @@ data Async m t where
     AsyncThreadId :: Promise m t -> Async m (ThreadId m)
     Race :: m t -> m r -> Async m (Either t r)
     Link :: Promise m t -> Async m ()
+    UnsafeUnmask :: m a -> Async m a
```
Comment (gromakovsky) [inline networking/src/Mockable/Concurrent.hs]: What's the story about `UnsafeUnmask`? Why did you add it?

File: auxx/src/Repl.hs (hunk 2 above)
Comment (int-index) [reply auxx/src/Repl.hs]: It is intentional: when a command is interrupted with an async exception, we do not want the REPL to crash.

File: networking/src/Mockable/Concurrent.hs (hunk 4 above)
Comment (int-index) [reply networking/src/Mockable/Concurrent.hs]: It was a part of `Mockable Bracket` before, and it is used in some of `Mockable Async` combinators.

Comment (gromakovsky) [reply networking/src/Mockable/Concurrent.hs]: Ok.

File: networking/src/Mockable/Concurrent.hs (hunk 3 above)
Comment (int-index) [reply networking/src/Mockable/Concurrent.hs]: this is oops
